alf/algorithms/algorithm.py

```diff
@@ -544,10 +569,13 @@ def _add_params_to_optimizer(params, opt):
             assert id(child) != id(self), "Child should not be self"
             if isinstance(child, Algorithm):
                 params, child_handled = child._setup_optimizers_(param_to_name)
-                for m in child_handled:
-                    assert m not in handled, duplicate_error % param_to_name.get(
-                        m)
-                    handled[m] = 1
+                if child.force_params_visible_to_parent:
+                    params = list(set(params + child_handled))
```

Comment:
I am wondering the use of `set` here can cause issue for checkpoint. See a related PR https://github.com/HorizonRobotics/alf/pull/891

Reply:
Make sense. Updated the logic to preserve the order.